dask_cuda/cuda_worker.py

```diff
@@ -84,9 +86,15 @@ def __init__(
             raise ValueError("nthreads must be higher than 0.")
 
         # Set nthreads=1 when parsing mem_limit since it only depends on nprocs
-        memory_limit = parse_memory_limit(
-            memory_limit=memory_limit, nthreads=1, total_cores=nprocs
-        )
+        if DISTRIBUTED_GT_2022_11_1:
+            logger = logging.getLogger(__name__)
+            memory_limit = parse_memory_limit(
+                memory_limit=memory_limit, nthreads=1, total_cores=nprocs, logger=logger
+            )
+        else:
+            memory_limit = parse_memory_limit(
+                memory_limit=memory_limit, nthreads=1, total_cores=nprocs
+            )
```

Comment:
I'd suggest we keep things simple, we don't want to keep this in the long-run, since we don't provide backwards compatibility anyway. For that reason I suggest the following instead:

```python
from distributed.utils import has_arg

if has_arg(parse_memory_limit, "logger"):
    logger = logging.getLogger(__name__)
    memory_limit = ...
else:
    memory_limit = ...
```

We can then drop the version compatibility code. Plus, I'd add a `TODO` comment or an issue to remove that in 23.02, when we'll definitely not support the old `2022.11.1` behavior.

Reply:
TIL that this utility exists.. Done, this is quite simpler too Peter.